marge/single_merge_job.py

```diff
@@ -92,9 +92,7 @@ def update_merge_request_and_accept(self, approvals):
 
             self.ensure_mergeable_mr(merge_request)
 
-            auto_squash = None
-            if self.is_auto_squash_enabled(merge_request):
-                auto_squash = True
+            auto_squash = True if target_project.squash_enforced else None
```

Comment:
Again from my previous comment, a user is allowed to choose if he / she wants squash at 3 different places.
* in the project settings
* in the merge request settings
* while merging the merge request

Out of all the case only one case we had to solve for when squash is set to `required` at project level. In this case we are supposed to pass this extra parameter `squash: True` which more or less actually seems like a bug to me from `gitlab` side, remaining cases we are not sending this parameter and letting gitlab take the decision based on it's business logic based on user's input.

Hence a distinction between False & None. It has been sometime since I have worked in python, so not sure if this the best way to write it. But this is intentional, I can add a comment over this line to make it more clear.